src/lib/y2storage/used_storage_features.rb

```diff
@@ -0,0 +1,208 @@
+#!/usr/bin/env ruby
+#
+# encoding: utf-8
+
+# Copyright (c) [2016-2017] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "y2storage"
+require "set"
+
+module Y2Storage
+  #
+  # Class that collects information about which storage features are used
+  # in the current target machine's storage setup so add-on packages can be
+  # marked for installation as needed.
+  #
+  # Usage:
+  #
+  #   # Create or obtain a device graph, e.g. with
+  #   devicegraph = Y2Storage::StorageManager.instance.probed
+  #
+  #   used_features = Y2Storage::UsedStorageFeatures.new(devicegraph)
+  #   pkg_list = used_features.feature_packages
+  #
+  class UsedStorageFeatures
+    include Yast::Logger
+
+    #======================================================================
+    # Configurable part starts here
+    #
+    # Software packages required for storage features.
+    # Any map value may be a string, a list of strings, or 'nil'.
+    #
+    # Packages that are part of a minimal installation (e.g., "util-linux")
+    # are not listed here.
+    #
+    # Those features correspond directly to the enum defined in
+    # https://github.com/openSUSE/libstorage-ng/blob/master/storage/UsedFeatures.h
+    #
+    FEATURE_PACKAGES =
+      {
+        # SUSE standard technologies
+        UF_LVM:           "lvm2",
+        # Btrfs needs e2fsprogs for 'lsattr' and 'chattr' to check for CoW
+        UF_BTRFS:         ["btrfsprogs", "e2fsprogs"],
+        UF_SNAPSHOTS:     "snapper",
+
+        # RAID technologies and related
+        UF_DM:            ["device-mapper"],
```

Comment:
This might be a leftover from times when that feature maybe required multiple packages. Or it might be just an accident.